packages/jest-core/src/SearchSource.ts

```diff
@@ -83,9 +83,11 @@ const globsToMatcher = (globs: Array<Config.Glob>) => {
   };
 };
 
-const regexToMatcher = (testRegex: Config.ProjectConfig['testRegex']) => (
-  path: Config.Path,
-) => testRegex.some(testRegex => new RegExp(testRegex).test(path));
+const regexToMatcher = (testRegex: Config.ProjectConfig['testRegex']) => {
+  const regexes = testRegex.map(testRegex => new RegExp(testRegex));
+
+  return (path: Config.Path) => regexes.some(regex => regex.test(path));
```

Comment:
stateful regexes might cause this to fail, no? not sure if it matters or not

Reply:
Ah yes that could be a problem. We can fix this by resetting lastIndex every time. Also not sure if that matters, but might as well do it to be safe.